test/unit/extensions/templateHander.tests.ts

```diff
@@ -0,0 +1,37 @@
+import { TemplateHandler, TemplateContext, ScopeData, TemplateData, Docx } from "src";
+import TemplateExtension, { mockExecute } from '../../__mocks__/TemplateExtension';
+
+beforeEach(() => {
+    TemplateExtension.mockClear();
+    mockExecute.mockClear();
+});
+
+describe(nameof(TemplateHandler), () => {
+    describe(nameof(TemplateHandler.prototype.callExtensions), () => {
```

Comment:
You should not unit test private methods. Instead just `describe(nameof(TemplateHandler.prototype.process))`.

Reply:
Makes sense now the method is private.